From e402ce0129e3d01c8c49a91b34a96e995d1e61ad Mon Sep 17 00:00:00 2001
From: Tanya Singh <tanya_singh@accton.com>
Date: Thu, 6 Aug 2026 11:17:52 +0800
Subject: [PATCH] mediatek: update Edgecore AP LED behavior

The LEDs were named green/orange/blue:indicator via color+function, so
board.d could only refer to them generically and the wiring did not
reflect the QSGs: EAP111 drove its uplink LED from eth1 (lan) and used
the blue LED for 5 GHz WLAN traffic, and no board indicated power.

Give each LED a functional label and wire it to the state its QSG
describes.

EAP111 / EAP111e:
- label the LEDs green:power, orange:ethernet, blue:cloud
- bind uplink activity to eth0 (wan); it tracked eth1 (lan)
- drop the 5 GHz WLAN trigger from blue, which the QSG reserves for
  cloud-managed state

EAP112:
- label the LEDs green:power, orange:halow, blue:lte, matching its
  three-LED IoT gateway scheme

EAP115 / EAP115a:
- label the LEDs green:power, orange:ethernet, blue:cloud
- keep uplink activity on eth0 (wan)

Signed-off-by: Tanya Singh <tanya_singh@accton.com>
---
 .../linux/mediatek/dts/mt7981a-edgecore-eap111.dts   |  3 +++
 .../linux/mediatek/dts/mt7981a-edgecore-eap112.dts   |  3 +++
 .../linux/mediatek/dts/mt7987a-edgecore-eap115.dtsi  |  3 +++
 .../mediatek/filogic/base-files/etc/board.d/01_leds  | 12 +++++++++---
 4 files changed, 18 insertions(+), 3 deletions(-)

diff --git a/target/linux/mediatek/dts/mt7981a-edgecore-eap111.dts b/target/linux/mediatek/dts/mt7981a-edgecore-eap111.dts
index 378ca13b5f..f7a6cf565c 100644
--- a/target/linux/mediatek/dts/mt7981a-edgecore-eap111.dts
+++ b/target/linux/mediatek/dts/mt7981a-edgecore-eap111.dts
@@ -36,18 +36,21 @@
 		compatible = "gpio-leds";
 
 		led_green: led-green {
+			label = "green:power";
 			gpios = <&pio 9 GPIO_ACTIVE_HIGH>;
 			color = <LED_COLOR_ID_GREEN>;
 			function = LED_FUNCTION_INDICATOR;
 		};
 
 		led_orange: led-orange {
+			label = "orange:ethernet";
 			gpios = <&pio 34 GPIO_ACTIVE_HIGH>;
 			color = <LED_COLOR_ID_ORANGE>;
 			function = LED_FUNCTION_INDICATOR;
 		};
 
 		led_blue: led-blue {
+			label = "blue:cloud";
 			gpios = <&pio 35 GPIO_ACTIVE_HIGH>;
 			color = <LED_COLOR_ID_BLUE>;
 			function = LED_FUNCTION_INDICATOR;
diff --git a/target/linux/mediatek/dts/mt7981a-edgecore-eap112.dts b/target/linux/mediatek/dts/mt7981a-edgecore-eap112.dts
index 8298e1aa72..047d360244 100644
--- a/target/linux/mediatek/dts/mt7981a-edgecore-eap112.dts
+++ b/target/linux/mediatek/dts/mt7981a-edgecore-eap112.dts
@@ -36,18 +36,21 @@
 		compatible = "gpio-leds";
 
 		led_green: led-green {
+			label = "green:power";
 			gpios = <&pio 9 GPIO_ACTIVE_HIGH>;
 			color = <LED_COLOR_ID_GREEN>;
 			function = LED_FUNCTION_INDICATOR;
 		};
 
 		led_orange: led-orange {
+			label = "orange:halow";
 			gpios = <&pio 34 GPIO_ACTIVE_HIGH>;
 			color = <LED_COLOR_ID_ORANGE>;
 			function = LED_FUNCTION_INDICATOR;
 		};
 
 		led_blue: led-blue {
+			label = "blue:lte";
 			gpios = <&pio 35 GPIO_ACTIVE_HIGH>;
 			color = <LED_COLOR_ID_BLUE>;
 			function = LED_FUNCTION_INDICATOR;
diff --git a/target/linux/mediatek/dts/mt7987a-edgecore-eap115.dtsi b/target/linux/mediatek/dts/mt7987a-edgecore-eap115.dtsi
index ba67e05354..8b1e9e8011 100644
--- a/target/linux/mediatek/dts/mt7987a-edgecore-eap115.dtsi
+++ b/target/linux/mediatek/dts/mt7987a-edgecore-eap115.dtsi
@@ -34,18 +34,21 @@
 		compatible = "gpio-leds";
 
 		led_green: led-green {
+			label = "green:power";
 			gpios = <&pio 9 GPIO_ACTIVE_HIGH>;
 			color = <LED_COLOR_ID_GREEN>;
 			function = LED_FUNCTION_INDICATOR;
 		};
 
 		led_orange: led-orange {
+			label = "orange:ethernet";
 			gpios = <&pio 12 GPIO_ACTIVE_HIGH>;
 			color = <LED_COLOR_ID_ORANGE>;
 			function = LED_FUNCTION_INDICATOR;
 		};
 
 		led_blue: led-blue {
+			label = "blue:cloud";
 			gpios = <&pio 8 GPIO_ACTIVE_HIGH>;
 			color = <LED_COLOR_ID_BLUE>;
 			function = LED_FUNCTION_INDICATOR;
diff --git a/target/linux/mediatek/filogic/base-files/etc/board.d/01_leds b/target/linux/mediatek/filogic/base-files/etc/board.d/01_leds
index fef7297f19..067d9b61ec 100644
--- a/target/linux/mediatek/filogic/base-files/etc/board.d/01_leds
+++ b/target/linux/mediatek/filogic/base-files/etc/board.d/01_leds
@@ -95,13 +95,19 @@ cudy,wbr3000uax-v1-ubootmod)
 	ucidef_set_led_netdev "lan" "lan" "blue:lan" "br-lan" "link tx rx"
 	;;
 edgecore,eap111|\
+edgecore,eap111e)
+	ucidef_set_led_netdev "wan" "wan" "orange:ethernet" "eth0"
+	ucidef_set_led_default "power" "POWER" "green:power" "on"
+	;;
 edgecore,eap112)
-	ucidef_set_led_netdev "wan" "wan" "orange:indicator" "eth1"
-	ucidef_set_led_netdev "wlan5g" "WLAN5G" "blue:indicator" "phy1-ap0"
+	# ucidef_set_led_netdev "halow" "HaLow" "orange:halow" "phy0-ap0"
+	# ucidef_set_led_netdev "wwan" "LTE" "blue:lte" "wwan0"
+	ucidef_set_led_default "power" "POWER" "green:power" "on"
 	;;
 edgecore,eap115|\
 edgecore,eap115a)
-	ucidef_set_led_netdev "wan" "wan" "orange:indicator" "eth0"
+	ucidef_set_led_netdev "wan" "wan" "orange:ethernet" "eth0"
+	ucidef_set_led_default "power" "POWER" "green:power" "on"
 	;;
 emplus,wap588m)
 	ucidef_set_led_netdev "lan" "LAN" "green:lan" "eth0"
-- 
2.34.1

